generate-xml/domain.nix

```diff
@@ -365,7 +365,11 @@ let
                 ]
                 [
                   (subelem "driver" [ (subattr "name" typeString) (subattr "model" typeString) ] [ ])
-                  (subelem "source" [ ] [ addresselem ])
+                  (subelem "source" [ ] [
+                    (subelem "vendor" [ (subattr "id" typeString) ] [ ])
+                    (subelem "product" [ (subattr "id" typeString) ] [ ])
+                    addresselem
```

Comment:
`id` for `vendor` and `product` should be `typeInt`, since they are actually numeric.